### PR TITLE
rbac: generate permission constants

### LINKED ISSUE #49075: Automatically generate permissions constants (for Go and TS) via sg generate

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #49075

This PR adds a `gen.go` file to generate permission constants on both backend and frontend whenever an update is made to the `schema.yaml` file.

## Test plan

* Manually tested using the `sg generate` command.
*

### CODE REVIEW

File: internal/rbac/yamldata.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	fmt.Fprintln(output)
+	fmt.Fprintf(output, "package %s\n", *pkgName)
+	for _, permission := range permissions {
+		fmt.Fprintln(output)
+		dn := permission.DisplayName()
+		fmt.Fprintln(output, fmt.Sprintf("const %sPermission string = \"%s\"", sentencizePermission(dn), dn))
+	}
+}
+
+var zanzibarPermissionRegex = regexp.MustCompile("([A-Za-z]+)(?:_([A-Za-z]+))?#([A-Za-z]+)")
```
Comment (courier-new) [inline internal/rbac/yamldata.go:59]: Is this used for anything?

**Review by courier-new — APPROVED** [review]
Nice! Do you mind adding some inline documentation about what this is generating?

Also, thoughts on how to get the same for clientside TS types? Could we reuse this same generator and supply another arg like `lang` to output in TS instead of a Go package?

File: internal/rbac/yamldata.go (hunk 1 above)
Comment (BolajiOlajide) [reply internal/rbac/yamldata.go:59]: No, I'll remove it.

File: internal/rbac/gen.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package rbac
+
+//go:generate env GO111MODULE=on go run yamldata.go -pkg rbac -o constants.go -lang go
+//go:generate gofmt -s -w constants.go
+//go:generate env GO111MODULE=on go run yamldata.go -pkg rbac -o ../../client/web/src/rbac/constants.ts -lang ts
```
Comment (courier-new) [inline internal/rbac/gen.go:5]: Nit: Probably don't need the `-pkg` arg on this one:
```suggestion
//go:generate env GO111MODULE=on go run yamldata.go -o ../../client/web/src/rbac/constants.ts -lang ts
```

File: internal/rbac/yamldata.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	case "go":
+		generateGoConstants(output, permissions)
+	case "ts":
+		generateTSConstants(output, permissions)
+	}
+}
+
+func generateTSConstants(output *os.File, permissions []types.Permission) {
+	fmt.Fprintln(output, "// Code generated by internal/rbac/yamldata. DO NOT EDIT.")
+	// export const Batc: string = 'dsdd'
```
Comment (courier-new) [inline internal/rbac/yamldata.go:62]:
```suggestion
```

**Review by courier-new — APPROVED** [review]
Amazinggg 😄 Only thought it, maybe we should backport it, just to prevent merge conflicts with ourselves as we do more QA/bug-fixing of RBAC?

**Pull request merged**